pkg/asset/ignition/node.go

```diff
@@ -91,3 +93,59 @@ func ConvertToAppendix(file *igntypes.File) {
 	file.Contents = igntypes.Resource{}
 	file.Overwrite = ignutil.BoolToPtr(false)
 }
+
+// parseCertificateBundle loads each certificate in the bundle to the Ingition
+// carrier type, ignoring any invisible character before, after and in between
+// certificates.
+func parseCertificateBundle(userCA []byte) ([]igntypes.Resource, error) {
+	userCA = bytes.TrimSpace(userCA)
+
+	var carefs []igntypes.Resource
+	for len(userCA) > 0 {
+		var block *pem.Block
+		block, userCA = pem.Decode(userCA)
+		if block == nil {
+			return nil, fmt.Errorf("unable to parse certificate, please check the certificates")
+		}
+
+		carefs = append(carefs, igntypes.Resource{Source: ignutil.StrToPtr(dataurl.EncodeBytes(pem.EncodeToMemory(block)))})
+
+		userCA = bytes.TrimSpace(userCA)
+	}
+
+	return carefs, nil
+}
+
+// GenerateIgnitionShim is used to generate an ignition file that contains a user ca bundle
+// in its Security section.
+func GenerateIgnitionShim(bootstrapConfigURL string, userCA string) ([]byte, error) {
```

Comment:
this moves the AWS ignition shim into a reusable function

Reply:
@bd233 yes this is good. Sorry for the confusing comment. I meant that as a note to myself.